tests/integration/test_fim/test_files/test_audit_buffer/test_audit_buffer_configuration.py

```diff
@@ -0,0 +1,270 @@
+
+    tier: 1
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration values for ossec.conf.
+        - metadata:
+            type: dict
+            brief: Test case data.
+        - test_folders:
+            type: dict
+            brief: List of folders to be created for monitoring.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set ossec.conf configuration.
+        - create_monitored_folders_module:
+            type: fixture
+            brief: Create a given list of folders when the module starts. Delete the folders at the end of the module.
+        - configure_local_internal_options_function:
+            type: fixture
+            brief: Set local_internal_options.conf file.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+
+    assertions:
+        - Verify when queue is full an event informs audit events may be lost
+        - Verify when queue is full at start up audit healthcheck fails and does not start
+        - Verify when using invalid values an error message is shown and does not start
+        - Verify configured queue_size value
+        - Verify real-time whodata thread is started correctly
+
+    input_description: The file 'configuration_audit_buffer_values' provides the configuration template.
+                       The file 'cases_audit_buffer_values.yaml' provides the tes cases configuration
+                       details for each test case.
+
+    expected_output:
+        - r".*(Internal audit queue is full). Some events may be lost. Next scheduled scan will recover lost data."
+        - r".*(Audit health check couldn't be completed correctly)."
+        - fr".*Invalid value for element (\'{element}\': .*)"
+        - r".*Internal audit queue size set to \'(.*)\'."
+        - r'.*File integrity monitoring (real-time Whodata) engine started.*'
+    '''
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    if metadata['valid_range']:
+        # Detect configured value
+        configured_value = get_configured_whodata_queue_size(wazuh_log_monitor)
+        assert str(metadata['queue_size']) in configured_value, 'Unexpected value found in "queue_size" in ossec.conf'
+
+    if not metadata['audit_starts']:
+        # Detect cause of failure
+        if metadata['fail_reason'] == 'queue_full':
+            detect_audit_queue_full(wazuh_log_monitor)
+            detect_audit_healthcheck_failed(wazuh_log_monitor)
+        elif metadata['fail_reason'] == 'invalid_value':
+            detect_invalid_conf_value(wazuh_log_monitor, element='queue_size')
+        with pytest.raises(TimeoutError):
+            # Detect real-time whodata thread does not start
+            detect_whodata_start(wazuh_log_monitor, timeout=T_5)
+    else:
+        # Detect real-time whodata thread started correctly
+        detect_whodata_start(wazuh_log_monitor)
```

Comment:
It validates only whodata. the log states 'real-time whodata thread', that is why the word realtime was used in the comment. removed to avoid confusion